### DISCUSSION TITLE
Experiment data storage location

### DISCUSSION THREAD

**Discussion by ghost** [jspsych/jsPsych#804] [Q&A] [answered]
Where is the data from a completed experiment saved to when using jsPsych?

**Comment by vijaymarupudi** [comment]
This link has more information: https://www.jspsych.org/overview/data/

Generally, it is not saved permanently by default. When the browser tab is closed, the data is lost. You will have to either save the data after the experiment to a file, using [`localSave()`](https://www.jspsych.org/core_library/jspsych-data/#localsave) or a server, following the example on the first link.

**Comment by jodeleeuw** [comment, marked as answer]
Hi @milsandhills,

That depends on what platform you are using to host the experiment. jsPsych runs entirely on the participant's machine, so all of the data is in the browser's memory on the participant's computer. You can send this data to the web server that is hosting the experiment. To be maximally flexible, jsPsych doesn't handle the web server component. This makes it compatible with a wide range of hosting services and tools:

* [cognition.run](http://cognition.run) - new free service with an easy user interface. great for starting off.
* [JATOS](https://www.jatos.org/Whats-JATOS.html) - a set of tools for hosting your own server. lots of flexibility.
* [pavlovia](https://pavlovia.org/) - paid hosting service for web-based experiments.
* [psiturk](https://psiturk.org/) - host your experiment on your own machine using python. relatively easy for a diy option.
* [pushkin](https://pushkin-social-science-at-scale.readthedocs.io/en/latest/index.html) - set up your own virtual laboratory for massive online experiments.
* full diy - you can setup a standard web server and database and handle the communication yourself.

I'm probably missing other options too.
